Add DNS support for PWD instances

xetorthio opened the pull request:
:bell: @marcosnils

marcosnils commented on api.go:
```diff
@@ -119,3 +130,30 @@ func main() {
+		}
+		m.Answer = append(m.Answer, a)
+		w.WriteMsg(m)
+		log.Println(m)
```
Unnecessary print?

marcosnils commented on services/session.go:
```diff
@@ -315,7 +318,10 @@ func LoadSessionsFromDisk() error {
 			// Connect PWD daemon to the new network
-			if err := ConnectNetwork("pwd", s.Id); err != nil {
+			if s.PwdIpAddress == "" {
+				log.Fatal("Cannot load stored sessions as they don't have the pwd ip address stored with them")
```
This Will fail when we deploy the new version right? Because none of the current sessions Will have that field. Also, is log.Fatal the best thing to do here? Maybe it's best to just avoid this session and remove it from the sessions map

xetorthio replied:
Yes. This is a safeguard. If it fails to load it's because an old version of the saved sessions file is used and should be removed.
This way we don't have mixed sessions.

marcosnils commented on api.go:
```diff
@@ -119,3 +130,29 @@ func main() {
+		// this is something we know about and we should try to handle
+		question := r.Question[0].Name
+		domainChunks := strings.Split(question, ".")
+		tldChunks := strings.Split(strings.TrimPrefix(domainChunks[0], "ip"), "-")
```
Question: "-" might now always come right?. Remember that we can also handle stuff like ip10_0_0_1.play-with-docker.com:8080 for which this `split` will fail.

xetorthio commented:
It won't fail! It will return the first value, which is the entire string.

On Fri, Feb 17, 2017, 07:57 Marcos Nils <notifications@github.com> wrote:

> *@marcosnils* commented on this pull request.
> ------------------------------
>
> In api.go
> <https://github.com/franela/play-with-docker/pull/94#discussion_r101733602>
> :
>
> > @@ -119,3 +130,29 @@ func main() {
>  	}
>  	log.Fatal(s.ListenAndServeTLS("", ""))
>  }
> +
> +func handleDnsRequest(w dns.ResponseWriter, r *dns.Msg) {
> +	if len(r.Question) > 0 && strings.HasSuffix(r.Question[0].Name, ".play-with-docker.com.") && strings.HasPrefix(r.Question[0].Name, "ip") {
> +		// this is something we know about and we should try to handle
> +		question := r.Question[0].Name
> +		domainChunks := strings.Split(question, ".")
> +		tldChunks := strings.Split(strings.TrimPrefix(domainChunks[0], "ip"), "-")
>
> Question: "-" might now always come right?. Remember that we can also
> handle stuff like ip10_0_0_1.play-with-docker.com:8080 for which this
> split will fail.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/franela/play-with-docker/pull/94#pullrequestreview-22471388>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIhp4wxsknJGoY2l-2_Jh0_nLOWfr5Iks5rdX0vgaJpZM4MDYtv>
> .
>

marcosnils commented:
> It won't fail! It will return the first value, which is the entire string.

right, I didn't see the split above. LGTM, we just need to fix the `domain` thing and we're good to go!.
